### PR TITLE
Add PhantomJS

### PR DESCRIPTION
**Pull request by joshuataylor**
Adds PhantomJS, which uses gastonjs.

### CODE REVIEW

File: php/phantomjs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+RUN cd /tmp && \
+    echo "f8afc8a24eec34c2badccc93812879a3d6f2caf3  phantomjs-2.1.1-linux-x86_64.tar.bz2" > phantomjs.sha1 && \
+    wget https://cnpmjs.org/mirrors/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2 && \
+    sha1sum -c phantomjs.sha1 && \
+    tar -xvf phantomjs-2.1.1-linux-x86_64.tar.bz2 && \
+    mv phantomjs-2.1.1-linux-x86_64/bin/phantomjs /usr/local/bin && \
+    rm phantomjs-2.1.1-linux-x86_64.tar.bz2
+
+RUN composer require behat/mink:~1.7 && \
```
Comment (nickschuch) [inline php/phantomjs/Dockerfile]: Maybe we can just checkout the git repo and point to that .js file with the phantomjs line?

File: php/phantomjs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+FROM previousnext/php:7.1
```
Comment (nickschuch) [inline php/phantomjs/Dockerfile]: Thoughts on using debian or alpine for base image?

File: php/phantomjs/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+WORKDIR /usr/local
+
+# bzip2 is required for phantomjs extract, libfontconfig for phantomjs
+RUN apt-get update && \
+    apt-get install -y bzip2 libfontconfig && \
+    apt-get clean
+
+RUN cd /tmp && \
+    echo "f8afc8a24eec34c2badccc93812879a3d6f2caf3  phantomjs-2.1.1-linux-x86_64.tar.bz2" > phantomjs.sha1 && \
```
Comment (nickschuch) [inline php/phantomjs/Dockerfile]: Thoughts on moving the hash + version to ENV declaration?

File: php/phantomjs/Dockerfile (hunk 1 above)
Comment (joshuataylor) [inline php/phantomjs/Dockerfile]: :+1:

File: php/phantomjs/Dockerfile (hunk 2 above)
Comment (joshuataylor) [inline php/phantomjs/Dockerfile]: :+1:

File: php/phantomjs/Dockerfile (hunk 3 above)
Comment (joshuataylor) [inline php/phantomjs/Dockerfile]: Good idea! Done.

**Comment by joshuataylor** [comment]
Made required changes.

**Comment by kimpepper** [comment]
As this no longer has anything to do with PHP, can we move these files under `phantomjs`?

File: php/phantomjs/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+WORKDIR /usr/local
+
+RUN apt-get update && \
+    apt-get install -y wget bzip2 libfontconfig git && \
+    apt-get clean
+
+RUN cd /tmp && \
+    echo $PHANTOMJS_HASH > phantomjs.sha1 && \
+    wget https://cnpmjs.org/mirrors/phantomjs/phantomjs-$PHANTOMJS_VERSION-linux-x86_64.tar.bz2 && \
+    sha1sum -c phantomjs.sha1 && \
```
Comment (nickschuch) [inline php/phantomjs/Dockerfile]: oh! does this have to be a file?

**Comment by nickschuch** [comment]
Under directory structure `phantomjs/2.1.1` and you have a deal :)

File: php/phantomjs/Dockerfile (hunk 4 above)
Comment (joshuataylor) [inline php/phantomjs/Dockerfile]: Can you use sha1sum to check a string against a file? I couldn't find anything under `man sha1sum`.

**Comment by kimpepper** [comment]
LGTM 👍

File: phantomjs/2.1.1/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+RUN cd /tmp && \
+    echo $PHANTOMJS_HASH > phantomjs.sha1 && \
+    wget https://cnpmjs.org/mirrors/phantomjs/phantomjs-$PHANTOMJS_VERSION-linux-x86_64.tar.bz2 && \
+    sha1sum -c phantomjs.sha1 && \
+    tar -xvf phantomjs-$PHANTOMJS_VERSION-linux-x86_64.tar.bz2 && \
+    mv phantomjs-$PHANTOMJS_VERSION-linux-x86_64/bin/phantomjs /usr/local/bin && \
+    rm phantomjs-$PHANTOMJS_VERSION-linux-x86_64.tar.bz2
+
+RUN git clone --depth=1 https://github.com/jcalderonzumba/gastonjs /data/gastonjs
```
Comment (benjy) [inline phantomjs/2.1.1/Dockerfile]: Do we need to be cloning a specific version here? Drupal actually has a version of this tagged, but if we do this here it won't come from the Drupal composer file, could lead to bugs in the future?

File: gastonjs/1.0.2/README.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+PhantomJS (GastonJS)
+===========================
+
+Installs PhantomJS, powered by [GasonJS](https://github.com/jcalderonzumba/gastonjs).
+
+For more information see this parent containers docuemntation.
```
Comment (benjy) [inline gastonjs/1.0.2/README.md]: s/docuemntation/documentation :)

File: phantomjs/2.1.1/Dockerfile (hunk 5 above)
Comment (joshuataylor) [inline phantomjs/2.1.1/Dockerfile]: :+1:. Using wget now to download the exact release, which also saves 70MB off the release!

**Comment by benjy** [comment]
LGTM

**Pull request merged**